### PR TITLE
Harvest expose schedule

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR expose the harvest scheduling into the API and the admin interface.

## API
The API expose 2 more entries:
- `POST /harvest/source/{ident}/schedule`: Schedule an harvest source. Expect a cron expression as payload
- `DELETE /harvest/source/{ident}/schedule`: Unschedule an harvest source

## Admin
The scheduling is displayed on the harvest source admin pages:
screenshot-www data dev-2017-09-11-18-56-21

The admin users can see a "Scheduling" menu entry on harvest source pages:
screenshot-www data dev-2017-09-11-18-36-19

On click, if the source is not scheduled, a default schedule is given (`'0 0 * * *'`). The admin user can:
- change it and submit by clicking on "Schedule"
- close the modal with the cross or the "Cancel button"

screenshot-www data dev-2017-09-11-18-42-25

If the harvest source is already scheduled, the existing schedule is displayed and the user can:
- change it and submit by clicking on "Schedule"
- close the modal with the cross or the "Cancel" button
- unschedule it using the "Unschedule" button

screenshot-www data dev-2017-09-11-18-42-53

The "Unschedule" button needs a confirmation so the modal change color (red for confirmation) and the user can:
- confirm using the "Confirm" button (perform the unschedule and close the modal)
- go back to the initial schedule form with the "Cancel" button

screenshot-www data dev-2017-09-11-18-43-16

Every action is confirmed with a notification:
screenshot-www data dev-2017-09-11-18-44-46
screenshot-www data dev-2017-09-11-18-46-55

The schedule display is updated accordingly:

screenshot-www data dev-2017-09-11-18-46-25

### CODE REVIEW

**Comment by abulte** [comment]
@noirbizarre Nice! A few comments before looking at the code:
- maybe we should link to or display an explanation text for the cron format? Maybe link to https://crontab.guru
- from my understanding the harvester is by default *not* scheduled and thus won't run. Isn't it dangerous — i.e. won't some organizations forget to schedule it?

**Comment by noirbizarre** [comment]
:+1: for the cron explanation.

The harvester is automatically scheduled on validation since udata 1.1.6 with #1114 
Beside, right now, only admin users can (re)schedule it.

File: js/components/harvest/schedule-modal.vue [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import API from 'api';
+
+import HorizontalForm from 'components/form/horizontal-form.vue';
+
+import HarvestSource from 'models/harvest/source';
+
+import Modal from 'components/modal.vue'
+
+const MASK = ['id', 'name', 'schedule'];
+const DEFAULT_SCHEDULE = '0 0 * * *';
```
Comment (abulte) [inline js/components/harvest/schedule-modal.vue]: Should not be fetched from the backend? Cf your first PR.

File: js/components/harvest/schedule-modal.vue [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        }
+    },
+    data() {
+        return {
+            source: new HarvestSource({mask: MASK}),
+            unscheduling: false,
+            fields: [{
+                id: 'schedule',
+                label: this._('Scheduling'),
+                description: this._('This is a cron expressions.')
```
Comment (abulte) [inline js/components/harvest/schedule-modal.vue]: expression

File: js/components/harvest/source.vue [hunk 3]
```diff
@@ -11,6 +11,9 @@
             <dl class="dl-horizontal">
                 <dt>{{ _('Backend') }}</dt>
                 <dd>{{ source.backend }}</dd>
+                <dt>{{ _('Scheduling') }}</dt>
+                <dd v-if="source.schedule">{{ source.schedule }}</dd>
+                <dd v-if="!source.schedule">{{ _('Not scheduled') }}</dd>
```
Comment (abulte) [inline js/components/harvest/source.vue]: `v-else`?

File: js/locales/udata.fr.json [hunk 4]
```diff
@@ -380,6 +381,7 @@
     "Unfollow": "Ne plus suivre",
     "Unique visitors": "Visiteurs uniques",
     "Unrecoverable error - this browser does not permit file uploading of any kind due to serious bugs in iOS8 Safari. Please use iOS8 Chrome until Apple fixes these issues.": "Erreur irrécupérable - ce navigateur ne permet l'envoie de fichier à cause d'un bug de Safari dans iOS8. veuillez utiliser Chrome pour iOS8 jusqu'à ce qu'Apple corrige ce problème.",
+    "Unschedule": "Deprogrammer",
```
Comment (abulte) [inline js/locales/udata.fr.json]: Déprogrammer

File: js/components/harvest/schedule-modal.vue (hunk 1 above)
Comment (noirbizarre) [reply js/components/harvest/schedule-modal.vue]: Yes, But i think I'll do that in another PR (needs an entrypoint or some configuration and this is only available to admin user)

**Comment by noirbizarre** [comment]
Done

**Pull request merged**